simpleviewinc/keg-core
# Zen 317 rollup config

**Ticket**: [ZEN-317](https://jira.simpleviewtools.com/browse/ZEN-317)

## Context/Goal

* to enable rollup export builds for taps, as needed by the events force team

## Updates

* `App.js`
  * fixed an incorrect import
* `core/base/components/kegComponents.js`
  * updated to a simpler import, which the rollup alias plugin ensures maps to the right .native export
* `core/base/hooks/route/useQuery.js`
  * updated an import to be more specific to avoid circular dependencies
* `core/base/store/index.js`
  * updated `devMiddleware` to only be included in a development environment and not the production build
* `core/configs/rollup.config.js`
  * import the tap's rollup config and export it here, so that rollup and run from the core's file path
* `package.json`
  * added roll:dev command, updated others, updated dependencies
* the remainder:
  * updated `jsutils` imports to use `@ltipton/jsutils`

## Testing

* `keg dpg run --package docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-317-rollup-export-truth`
  * this will start up `tap-events-force`, using this branch of core and [this branch](https://github.com/simpleviewinc/tap-resolver/pull/22) of `tap-resolver`
* navigate to `kegdev.xyz` in your browser, and verify that the tap runs as expected
* attach to the container (`keg d att`, then select the container)
* `yarn roll:dev`
* verify the build is created with no errors
* `keg dc`, then copy the container id of this package, and hold on to it for later
* if you haven't already, clone `keg-test-consumer` in your `~/keg` directory: https://github.com/simpleviewinc/keg-test-consumer
* inside of `keg-test-consumer`, run `yarn install`
* next, using the docker container id you captured earlier, run `yarn copy:build <container_id`
* now, run `yarn start`
* verify that sessions component appears and behaves as expected in the app window.
